scripts/simulateFunctionsJavaScript.js

```diff
@@ -1,12 +1,37 @@
-const { simulateRequest, getDecodedResultLog } = require("../FunctionsSandboxLibrary")
-const requestConfig = require("../Functions-request-config.js")
+const { simulateRequest, getDecodedResultLog, getRequestConfig } = require("../FunctionsSandboxLibrary")
 
-;(async () => {
+const runSimulation = async (requestConfig) => {
   const { resultLog, result, success } = await simulateRequest(requestConfig)
 
   console.log(`\n${resultLog}`)
   if (success) {
-    console.log(`Value returned from souce code: ${result}\n${getDecodedResultLog(requestConfig, result)}`)
+    console.log(`Value returned from source code: ${result}\n${getDecodedResultLog(requestConfig, result)}`)
     return
   }
+}
+
+(async () => {
+    const unvalidatedRequestConfig = require("../Functions-request-config.js")
+    const requestConfig = getRequestConfig(unvalidatedRequestConfig)
+
+    if (requestConfig.secretsLocation === 1) {
+      if (requestConfig.secrets && Object.keys(requestConfig.secrets).length !== 0) {
+        console.log('\n__SIMULATING JAVASCRIPT WITH DEFAULT SECRETS__'.toUpperCase())
+        await runSimulation(requestConfig)
+      }
+
+      if (!requestConfig.perNodeSecrets) {
+        return
+      }
+
+      for (let i = 0; i < requestConfig.perNodeSecrets.length; i++) {
+        requestConfig.secrets = requestConfig.perNodeSecrets[i]
+        console.log(`\n__SIMULATING JAVASCRIPT WITH SECRETS ASSIGNED TO NODE ${i}__`.toUpperCase())
+        await runSimulation(requestConfig)
+      }
+      return
+    }
+
+    console.log('\n__Simulating JavaScript__')
```

Comment:
And this is not uppercase.. I am wondering why some log messages are uppercase while some are not?
I think all tooling output shall be consistent.

Reply:
It was to easily identify which log statements belong to each each per-node execution.

Reply:
I personally find it useful, so I would advocate to keep it.  It is not a function that can be included in the HardHat plugin anyway since it is just an NPM command so a user can run a sanity check on their code.